### PR TITLE
Fix - `enable_alt_bn128_syscall`

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
The feature affects RBPF as it changes the set of valid syscalls, so it must be included in `FEATURES_AFFECTING_RBPF` as well.

#### Summary of Changes
- Adds `enable_alt_bn128_syscall` to `FEATURES_AFFECTING_RBPF`.
- Fixes `test_runtime_feature_enable_with_program_cache`.

### CODE REVIEW

File: runtime/src/bank/tests.rs
Comment (pgarg66) [inline runtime/src/bank/tests.rs:1]: Should we split the test fix as a separate PR?

Comment (Lichtso) [reply runtime/src/bank/tests.rs:1]: We can, but the test is related to `FEATURES_AFFECTING_RBPF` and both need to be backported.

**Review by pgarg66 — APPROVED** [review]
